common/src/main/java/io/netty/util/concurrent/Future.java

```diff
@@ -166,4 +168,46 @@
      */
     @Override
     boolean cancel(boolean mayInterruptIfRunning);
+
+    /**
+     * Returns the {@link EventExecutor} that is tied to this {@link Future}.
+     */
+    EventExecutor executor();
+
+    @Override
+    default V get() throws InterruptedException, ExecutionException {
+        await();
+
+        Throwable cause = cause();
+        if (cause == null) {
+            return getNow();
+        }
+        if (cause instanceof CancellationException) {
+            throw (CancellationException) cause;
+        }
+        throw new ExecutionException(cause);
+    }
+
+    @Override
+    default V get(long timeout, TimeUnit unit) throws InterruptedException, ExecutionException, TimeoutException {
+        if (await(timeout, unit)) {
+            Throwable cause = cause();
+            if (cause == null) {
+                return getNow();
+            }
+            if (cause instanceof CancellationException) {
+                throw (CancellationException) cause;
+            }
+            throw new ExecutionException(cause);
+        }
+        throw new TimeoutException();
+    }
+
+    /**
+     * Returns a {@link CompletionStage} that reflects the state of this {@link Future} and so will receive
```

Comment:
Nit: Add a `@link` to `CompletionStageAdapter`

Reply:
Introduced `FutureCompletionStage` to make semantics more clear